File: back-end/BookStore.Models/Models/UserModel.cs

using BookStore.Models.ViewModels;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BookStore.Models.Models
{
    public class UserModel
    {  public UserModel() { }
        public UserModel(User user)
        {
            id = user.Id;
            firstName = user.Firstname;
            lastName = user.Lastname;
            email = user.Email;
            roleId = user.Roleid;
        }
        public int id { get; set; }
        public string firstName { get; set; } = null!;
        public string lastName { get; set; } = null!;
        public string email { get; set; } = null!;
        public string password { get; set; } = null!;
        public int roleId { get; set; }
    }
}

File: Front-end/src/models/UserModel.ts
export class AddOrEditUserModel {
	id?: number;
	email!: string;
	firstName!: string;
	lastName!: string;
	roleId!: number;
	name?: string;
	password?: string;
}

export class UpdateProfileModel {
	email!: string;
	firstName!: string;
	lastName!: string;
	newPassword?: string;
	confirmPassword?: string;
}

export default class UserModel {
	id?: number;
	email!: string;
	firstName!: string;
	lastName!: string;
	roleId!: number;
	role?: string;
	password?: string;
}
